adr/component_structure.md

```diff
@@ -0,0 +1,102 @@
+# Create both low-level and high-level components that are exported from each package.
+
+## Context
+
+The primary issue we are facing right now is determining how "high-level" the component should be. For example, should we create a single component with many props, or should we create many components to be composed together.
+
+```jsx
+<ClayCard>
+	<ClayCard.Title onClick={handleTitleClick}>
+		<ClayIcon symbol="check" />
+
+		{'Hello World'}
+	</ClayCard.Title>
+
+	<ClayCard.Body onClick={handleBodyClick}>
+		<ClayCard.Description>
+			{'Some cool descriptors'}
+		</ClayCard.Description>
+	</ClayCard.Body>
+</ClayCard>
+
+// vs
+<ClayCard
+	iconTitleLeftSymbol="check"
+	onTitleClick={handleTitleClick}
+	onBodyClick={handleBodyClick}
+	description="foo bar baz"
+	title="hello world"
+/>
+```
+
+## Decision
+
+The decision going forward is to create 2 sets of components
+
+-   low-level composable components such as `<ClayCard.Header>`, `<ClayCard.Footer>`
+-   simple high-level components that cover basic lexicon uses
+
+### What does this look like?
+
+#### Composable:
+
+![Composable Drop Down](./images/component_structure_2.png)
+
+```jsx
+<ClayDropDown>
+	<ClayDropDown.Search
+		onChange={() => {}}
+		placeholder="Search"
+		spritemap={spritemap}
+		value={''}
+	/>
+
+	<ClayDropDown.ItemList>
+		{items.map((item, i) => (
+			<ClayDropDown.Item href={item.href} key={i}>
+				{item.label}
+			</ClayDropDown.Item>
+		))}
+	</ClayDropDown.ItemList>
+</ClayDropDown>
+```
+
+Composing gives the user more control over the layout and interactions.
+
+#### High-level:
+
+![High Level Drop Down](./images/component_structure_1.png)
+
+```jsx
+<ClayDropDownWithPagination
+	delta={7}
+	header="Favorites"
+	items={items} // array of objects with `label` and `href`
+	message={'Showing {0} of {1} Structures'}
+	title="Folder"
+/>
+```
```

Comment:
For the sake of fairness, I think all here should be mapped correctly to a corresponding composable example. I think that way it would be easier to spot the possible benefit or drawbacks of each approach for different cases.

```jsx
<ClayDropDown>
        <ClayDropDown.Title>Folder</ClayDropdown.Header>

 	<ClayDropDown.ItemList header="Favorites">
		{items.filter((item, i) => i < delta).map((item, i) => (
			<ClayDropDown.Item href={item.href} key={i}>
				{item.label}
			</ClayDropDown.Item>
		))}
	</ClayDropDown.ItemList>

 	{items.length > delta && 
 	 	<ClayDropDown.Pagination message="Showing {0} of {1} Structures">
 	 	 	 	<ClayButton label="More" onClick="{...}" />
 	 	</ClayDropDown.Pagination>
 	}
</ClayDropDown>
```